### PR TITLE
Output Recording will run for Microsoft Simulator

### PR DESCRIPTION
**Pull request by ScottCarda-MS**
Updates the LLVM's BitcodeModule to add hooks that are used by the Runtime to get EntryPoint information from the bit-code.

### CODE REVIEW

File: src/QsCompiler/QirGeneration/CompilationSteps.cs [hunk 1]
```diff
@@ -46,9 +46,17 @@ internal static bool VerifyGenerationPrecondition(QsCompilation compilation, Lis
...
         {
+            var targetCapability = assemblyConstants.TryGetValue(ReservedKeywords.AssemblyConstants.TargetCapability, out var capability) && !string.IsNullOrWhiteSpace(capability)
+                ? TargetCapability.TryParse(capability) // null if parsing fails
+                : null;
+            var isMicrosoftSimulator =
+                assemblyConstants.TryGetValue(ReservedKeywords.AssemblyConstants.ProcessorArchitecture, out var architecture)
+                && architecture == ReservedKeywords.AssemblyConstants.MicrosoftSimulator;
+
             compilation =
+                isMicrosoftSimulator ||
```
Comment (cesarzc) [inline src/QsCompiler/QirGeneration/CompilationSteps.cs:59]: I believe we might now be effectively applying `AddOutputRecording` for all cases where we generate QIR. This condition could go away and the transformation could always be applied.

Comment (ScottCarda-MS) [reply src/QsCompiler/QirGeneration/CompilationSteps.cs:59]: Should I also get rid of the part of the condition pertaining to the target capabilities?

Comment (cesarzc) [reply src/QsCompiler/QirGeneration/CompilationSteps.cs]: Yes, it seems like you already did that :)

**Review by cesarzc — APPROVED** [review]

**Pull request merged**